include/mbedtls/ecdh.h

```diff
@@ -63,6 +93,15 @@ typedef struct mbedtls_ecdh_context
     mbedtls_ecp_point Vi;    /*!< The blinding value. */
     mbedtls_ecp_point Vf;    /*!< The unblinding value. */
     mbedtls_mpi _d;          /*!< The previous \p d. */
+#else
+    int point_format;        /*!< The format of point export in TLS messages. */
+    mbedtls_ecp_group_id grp;/*!< The elliptic curve used. */
+    mbedtls_ecdh_variant var;/*!< The ECDH implementation/structure used. */
+    union
+    {
+        mbedtls_ecdh_context_mbed   *mbed_ecdh;
```

Comment:
Why is this a pointer? Do we expect the alternative contexts to have wildly differing sizes? A pointer adds memory management overhead, which impacts RAM usage, performance (especially wrt caches) code size and complexity (this is visible in `mbedtls_ecdh_setup_internal`). For RAM alone, how much overhead there is per block depends wildly on the malloc implementation, but it's hard to go below 8 bytes of overhead, plus the pointer makes 12 bytes, so it's a net loss unless the size difference between two context types can be above 12 bytes.

Reply:
From our end, both are fine. I will run some experiments to find out whether it makes a difference in our applications or not, but I don't expect a big difference.

Reply:
The context sizes are 64 vs 300 where the latter can be reduced to 216. That seems acceptable so based on this and your comments I'll remove the pointer.